### PR TITLE
[Routing] Document the PSR-4 route loader

### PR DESCRIPTION
**Pull request by derrabus**
This PR documents symfony/symfony#47916

Although the PR does not deprecate `AnnotationDirectoryLoader` and `AnnotationFileLoader`, I chose to recommend only the new `Psr4DirectoryLoader` and `AnnotationDirectoryLoader` in `routing.rst`. In `routing/custom_route_loader.rst`, all four loaders are listed for completeness.

Using a class name as resource instead of a file path has always worked, but wasn't documented for some reason. I have added examples for that as well.

### CODE REVIEW

**Review by welcoMattic — APPROVED** [review]

File: routing.rst
Comment (maxbeckers) [inline routing.rst:48]: missing colon `directory, which follows`?

File: routing.rst
Comment (maxbeckers) [inline routing.rst:49]: Here as well `act as a controller as well, which`?

File: routing.rst
Comment (derrabus) [reply routing.rst:48]: Not sure, it feels very German to set that comma. Let's see what the docs team says.

**Review by maxbeckers — APPROVED** [review]

**Comment by javiereguiluz** [comment]
I've tagged this with "Waiting code merge" because we're discussing internally about some potential tweaks to the syntax of this new feature. Let's wait for a decision before merging. Thanks.

**Comment by javiereguiluz** [comment]
Thanks Alexander!

Note: while merging I updated the config to match the changes in https://github.com/symfony/symfony/pull/47943

**Pull request merged**
